tests/test_tagbot.py

```diff
@@ -267,6 +268,38 @@ def test_handle_open():
         == "Approved and merged."
     )
     tagbot.client.projects.get.assert_called_once_with(1, lazy=True)
+    # Assert we've gotten the MR twice (the second time is just for printing)
+    calls = [call(None, lazy=True), call(None, lazy=False)]
+    p.mergerequests.get.assert_has_calls(calls)
+    mr.approve.assert_called_once_with()
+    mr.merge.assert_called_once_with(
+        merge_when_pipeline_succeeds=True, should_remove_source_branch=True
+    )
+
+    # Test with head pipeline not set
+    # Note it should still approve and try merging, GitLab API will throw an error if
+    # the merge is not valid but this is mocked so it succeeds fine
+    mr = Mock(spec=gitlab.v4.objects.ProjectMergeRequest)
```

Comment:
I'm not super familiar with `Mock`, does the `sleep` still activate when this stuff is mocked? you could perhaps mock `sleep` so it doesn't actually take 1 second (if it does)

Reply:
Thanks for the tip